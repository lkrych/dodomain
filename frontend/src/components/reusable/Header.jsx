import React, {Component} from 'react';  
import { Link } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';

const styles = theme => ({
  root: {
    flexGrow: 1,
  },
  flex: {
    flex: 1,
  },
  button: {
    margin: theme.spacing.unit,
  },
  link: {
    textDecoration: "none"
  }
});

class Header extends Component {  
  render() {
    let { classes } = this.props;
    if (this.props.logged_in) {
      return (
        <div className={classes.root}>
          <AppBar position="static">
            <Toolbar>
              <Typography variant="title" color="inherit" className={classes.flex}>
                  Dodomain
              </Typography>
              <Link to="/" className={classes.link} >
                <Button variant="contained" color="primary" className={classes.button}>
                  Home
                </Button>
              </Link>
              <Link to="/index" className={classes.link} >
                <Button variant="contained" color="primary" className={classes.button}>
                  Domain List
                </Button>
              </Link>
              <Link to="/submit" className={classes.link} >
                <Button variant="contained" color="primary" className={classes.button}>
                  Submit Domain
                </Button>
              </Link>
              <Link to="/" id="logout" className={classes.link}
                onClick={this.props.logout} >
                <Button variant="contained" color="primary" className={classes.button}>
                  Log Out
                </Button>
              </Link>
            </Toolbar>
          </AppBar>
        </div>
      );
    } else {
      return (
        <div className={classes.root}>
          <AppBar position="static">
            <Toolbar>
              <Typography variant="title" color="inherit" className={classes.flex}>
                  Dodomain
              </Typography>
              <Link to="/" className={classes.link} >
                <Button variant="contained" color="primary" className={classes.button}>
                  Home
                </Button>
              </Link>
              <Link to="/login" className={classes.link} >
                <Button variant="contained" color="primary" className={classes.button}>
                  Log In
                </Button>
              </Link>
              <Link to="/signup" className={classes.link} >
                <Button variant="contained" color="primary" className={classes.button}>
                  Sign Up
                </Button>
              </Link>
            </Toolbar>
          </AppBar>
        </div>
      );
    }
  }
}



export default withStyles(styles)(Header);