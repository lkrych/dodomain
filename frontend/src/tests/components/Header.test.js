import React from 'react';
import { shallow } from 'enzyme';

import Header from '../../components/reusable/Header';

describe('Header', () => {
  
  let loggedIn = shallow(<Header logged_in={true} />);
  let loggedOut = shallow(<Header logged_in={false} />);
  
  it('renders correctly', () => {
    expect(loggedOut).toMatchSnapshot();
  });

  describe ('changes what is displayed depending on if a user is signed in', () => {
    

    it ('should display the login link if a user is not signed in', () => {
      expect(loggedOut.find('#login').length).toBeGreaterThanOrEqual(1);
      expect(loggedOut.find('#logout').length).toEqual(0);
    });

    it ('should display the logout link if a user is signed in', () => {
      expect(loggedIn.find('#logout').length).toBeGreaterThanOrEqual(1);
      expect(loggedIn.find('#login').length).toEqual(0);
    });
  });

  describe ('when logged in', () => {

    it('clicking on the logout link', () => {
      beforeEach(() => {
        loggedIn.find('#logout').simulate('click'); //allows us to find inner child nodes or components by jsx tags or by their classname
      });

      it('should change the header', () => {
        expect(loggedIn.find('#login').length).toBeGreaterThanOrEqual(1);
        expect(loggedIn.find('#logout').length).toEqual(0);
      });
    });
  });

});
