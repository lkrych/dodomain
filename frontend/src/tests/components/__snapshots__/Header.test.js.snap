// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    classes={
        Object {
            "root": "blahStyle",
          }
    }
    logged_in={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(AppBar)
        position="static"
>
        <WithStyles(Toolbar)>
                <WithStyles(Typography)
                        className={undefined}
                        color="inherit"
                        variant="title"
                >
                        Dodomain
                </WithStyles(Typography)>
                <Link
                        className={undefined}
                        replace={false}
                        to="/"
                >
                        <WithStyles(Button)
                                className={undefined}
                                color="primary"
                                variant="contained"
                        >
                                Home
                        </WithStyles(Button)>
                </Link>
                <Link
                        className={undefined}
                        id="login"
                        replace={false}
                        to="/login"
                >
                        <WithStyles(Button)
                                className={undefined}
                                color="primary"
                                variant="contained"
                        >
                                Log In
                        </WithStyles(Button)>
                </Link>
                <Link
                        className={undefined}
                        replace={false}
                        to="/signup"
                >
                        <WithStyles(Button)
                                className={undefined}
                                color="primary"
                                variant="contained"
                        >
                                Sign Up
                        </WithStyles(Button)>
                </Link>
        </WithStyles(Toolbar)>
</WithStyles(AppBar)>,
      "className": "blahStyle",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Toolbar)>
          <WithStyles(Typography)
                    className={undefined}
                    color="inherit"
                    variant="title"
          >
                    Dodomain
          </WithStyles(Typography)>
          <Link
                    className={undefined}
                    replace={false}
                    to="/"
          >
                    <WithStyles(Button)
                              className={undefined}
                              color="primary"
                              variant="contained"
                    >
                              Home
                    </WithStyles(Button)>
          </Link>
          <Link
                    className={undefined}
                    id="login"
                    replace={false}
                    to="/login"
          >
                    <WithStyles(Button)
                              className={undefined}
                              color="primary"
                              variant="contained"
                    >
                              Log In
                    </WithStyles(Button)>
          </Link>
          <Link
                    className={undefined}
                    replace={false}
                    to="/signup"
          >
                    <WithStyles(Button)
                              className={undefined}
                              color="primary"
                              variant="contained"
                    >
                              Sign Up
                    </WithStyles(Button)>
          </Link>
</WithStyles(Toolbar)>,
        "position": "static",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              className={undefined}
              color="inherit"
              variant="title"
>
              Dodomain
</WithStyles(Typography)>,
            <Link
              className={undefined}
              replace={false}
              to="/"
>
              <WithStyles(Button)
                            className={undefined}
                            color="primary"
                            variant="contained"
              >
                            Home
              </WithStyles(Button)>
</Link>,
            <Link
              className={undefined}
              id="login"
              replace={false}
              to="/login"
>
              <WithStyles(Button)
                            className={undefined}
                            color="primary"
                            variant="contained"
              >
                            Log In
              </WithStyles(Button)>
</Link>,
            <Link
              className={undefined}
              replace={false}
              to="/signup"
>
              <WithStyles(Button)
                            className={undefined}
                            color="primary"
                            variant="contained"
              >
                            Sign Up
              </WithStyles(Button)>
</Link>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Dodomain",
              "className": undefined,
              "color": "inherit",
              "variant": "title",
            },
            "ref": null,
            "rendered": "Dodomain",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Button)
                className={undefined}
                color="primary"
                variant="contained"
>
                Home
</WithStyles(Button)>,
              "className": undefined,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Home",
                "className": undefined,
                "color": "primary",
                "variant": "contained",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Button)
                className={undefined}
                color="primary"
                variant="contained"
>
                Log In
</WithStyles(Button)>,
              "className": undefined,
              "id": "login",
              "replace": false,
              "to": "/login",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Log In",
                "className": undefined,
                "color": "primary",
                "variant": "contained",
              },
              "ref": null,
              "rendered": "Log In",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Button)
                className={undefined}
                color="primary"
                variant="contained"
>
                Sign Up
</WithStyles(Button)>,
              "className": undefined,
              "replace": false,
              "to": "/signup",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sign Up",
                "className": undefined,
                "color": "primary",
                "variant": "contained",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(AppBar)
          position="static"
>
          <WithStyles(Toolbar)>
                    <WithStyles(Typography)
                              className={undefined}
                              color="inherit"
                              variant="title"
                    >
                              Dodomain
                    </WithStyles(Typography)>
                    <Link
                              className={undefined}
                              replace={false}
                              to="/"
                    >
                              <WithStyles(Button)
                                        className={undefined}
                                        color="primary"
                                        variant="contained"
                              >
                                        Home
                              </WithStyles(Button)>
                    </Link>
                    <Link
                              className={undefined}
                              id="login"
                              replace={false}
                              to="/login"
                    >
                              <WithStyles(Button)
                                        className={undefined}
                                        color="primary"
                                        variant="contained"
                              >
                                        Log In
                              </WithStyles(Button)>
                    </Link>
                    <Link
                              className={undefined}
                              replace={false}
                              to="/signup"
                    >
                              <WithStyles(Button)
                                        className={undefined}
                                        color="primary"
                                        variant="contained"
                              >
                                        Sign Up
                              </WithStyles(Button)>
                    </Link>
          </WithStyles(Toolbar)>
</WithStyles(AppBar)>,
        "className": "blahStyle",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Toolbar)>
            <WithStyles(Typography)
                        className={undefined}
                        color="inherit"
                        variant="title"
            >
                        Dodomain
            </WithStyles(Typography)>
            <Link
                        className={undefined}
                        replace={false}
                        to="/"
            >
                        <WithStyles(Button)
                                    className={undefined}
                                    color="primary"
                                    variant="contained"
                        >
                                    Home
                        </WithStyles(Button)>
            </Link>
            <Link
                        className={undefined}
                        id="login"
                        replace={false}
                        to="/login"
            >
                        <WithStyles(Button)
                                    className={undefined}
                                    color="primary"
                                    variant="contained"
                        >
                                    Log In
                        </WithStyles(Button)>
            </Link>
            <Link
                        className={undefined}
                        replace={false}
                        to="/signup"
            >
                        <WithStyles(Button)
                                    className={undefined}
                                    color="primary"
                                    variant="contained"
                        >
                                    Sign Up
                        </WithStyles(Button)>
            </Link>
</WithStyles(Toolbar)>,
          "position": "static",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                className={undefined}
                color="inherit"
                variant="title"
>
                Dodomain
</WithStyles(Typography)>,
              <Link
                className={undefined}
                replace={false}
                to="/"
>
                <WithStyles(Button)
                                className={undefined}
                                color="primary"
                                variant="contained"
                >
                                Home
                </WithStyles(Button)>
</Link>,
              <Link
                className={undefined}
                id="login"
                replace={false}
                to="/login"
>
                <WithStyles(Button)
                                className={undefined}
                                color="primary"
                                variant="contained"
                >
                                Log In
                </WithStyles(Button)>
</Link>,
              <Link
                className={undefined}
                replace={false}
                to="/signup"
>
                <WithStyles(Button)
                                className={undefined}
                                color="primary"
                                variant="contained"
                >
                                Sign Up
                </WithStyles(Button)>
</Link>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Dodomain",
                "className": undefined,
                "color": "inherit",
                "variant": "title",
              },
              "ref": null,
              "rendered": "Dodomain",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Button)
                  className={undefined}
                  color="primary"
                  variant="contained"
>
                  Home
</WithStyles(Button)>,
                "className": undefined,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Home",
                  "className": undefined,
                  "color": "primary",
                  "variant": "contained",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Button)
                  className={undefined}
                  color="primary"
                  variant="contained"
>
                  Log In
</WithStyles(Button)>,
                "className": undefined,
                "id": "login",
                "replace": false,
                "to": "/login",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Log In",
                  "className": undefined,
                  "color": "primary",
                  "variant": "contained",
                },
                "ref": null,
                "rendered": "Log In",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Button)
                  className={undefined}
                  color="primary"
                  variant="contained"
>
                  Sign Up
</WithStyles(Button)>,
                "className": undefined,
                "replace": false,
                "to": "/signup",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Sign Up",
                  "className": undefined,
                  "color": "primary",
                  "variant": "contained",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
